source/go-perun/app_tutorial/client.rst

```diff
@@ -0,0 +1,329 @@
+            const assetIdx, peerIdx = 0, 1
+            if err := channel.AssetsAssertEqual(lcp.InitBals.Assets, []channel.Asset{c.currency}); err != nil {
+                return nil, fmt.Errorf("Invalid assets: %v\n", err)
+            } else if lcp.FundingAgreement[assetIdx][peerIdx].Cmp(c.stake) != 0 {
+                return nil, fmt.Errorf("Invalid funding balance")
+            }
+            return lcp, nil
+        }()
+        if err != nil {
+            r.Reject(context.TODO(), err.Error()) //nolint:errcheck // It's OK if rejection fails.
+        }
+
+As in the payment channel example, we create the accept message, send it, and start the on-chain watcher.
+
+.. code-block:: go
+
+        // Create a channel accept message and send it.
+        accept := lcp.Accept(
+            c.account,                // The account we use in the channel.
+            client.WithRandomNonce(), // Our share of the channel nonce.
+        )
+        ch, err := r.Accept(context.TODO(), accept)
+        if err != nil {
+            fmt.Printf("Error accepting channel proposal: %v\n", err)
+            return
+        }
+
+        // Start the on-chain event watcher. It automatically handles disputes.
+        c.startWatching(ch)
+
+        c.channels <- newAppChannel(ch)
+    }
+
+.. _app_client_channel:
+
+App channel
+-----------
+
+We implement the type ``AppChannel`` that wraps a Perun channel and provides convenient functions to interact with the app channel.
+We put this functionality in ``client/channel.go``.
+
+.. code-block:: go
+
+    type AppChannel struct {
+        ch *client.Channel
+    }
+
+.. _app_client_channel_set:
+
+Set
+...
+We create ``Set``, which expects ``x`` and ``y`` that indicate where the client wants to put its symbol on the grid.
+
+We use ``channel.UpdateBy`` for conveniently proposing our desired update to the channel's ``state``.
+With ``state.App``, we fetch the app that identifies the channel's application and check if it is of the expected type.
+Then we call ``TicTacToeApp.Set``, which will manipulate ``state.Data`` to include the desired turn for us.
+``state.Data`` holds all app-specific data, reflecting our Tic-Tac-Toe game's current game state.
+We will go into more detail about this at :ref:`a later stage <app_set>`.
+
+.. code-block:: go
+
+    func (g *AppChannel) Set(x, y int) {
+        err := g.ch.UpdateBy(context.TODO(), func(state *channel.State) error {
+            app, ok := state.App.(*app.TicTacToeApp)
+            if !ok {
+                return fmt.Errorf("invalid app type: %T", app)
+            }
+
+            return app.Set(state, x, y, g.ch.Idx())
+        })
+        if err != nil {
+            panic(err) // We panic on error to keep the code simple.
+        }
+    }
+
+Handling Update
+...............
+We don't need to do much here in the app channel case.
+Basically, we can accept every update we receive because the Perun app-channel automatically checks that the transition is valid.
```

Comment:
```suggestion
Basically, we can accept every update we receive because the app's :ref:`valid transition function <app_validate_transition>` already ensures that the transition is valid.
```
It's not Perun's app channel. It's the valid transition logic we defined.